iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/rules/ThrottleResistantTestRule.java

```diff
@@ -49,6 +50,19 @@ public void evaluate() throws Throwable {
                     log.warn("Thottling detected in test {}, waiting for {} milliseconds and then re-running the test", description.getMethodName(), THROTTLING_RETRY_DELAY_MILLISECONDS, e);
                     Thread.sleep(THROTTLING_RETRY_DELAY_MILLISECONDS);
                 }
+                catch (Exception e)
+                {
+                    if (e.getMessage() != null && e.getMessage().toLowerCase().contains(CONNECTION_REFUSED))
+                    {
+                        log.warn("Thottling detected in test {}, waiting for {} milliseconds and then re-running the test", description.getMethodName(), THROTTLING_RETRY_DELAY_MILLISECONDS, e);
```

Comment:
I only see 2 instances of `{}`, but 3 parameters. Is the exception parameter a special one?

Reply:
The exception parameter is indeed special here. The log.warn (and log.error) will handle this exception without needing a ```{}```